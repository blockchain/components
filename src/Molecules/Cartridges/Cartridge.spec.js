import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Cartridge from './Cartridge'

describe('Cartridge component', () => {
  it('renders correctly', () => {
    const component = shallow(<Cartridge />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with nature', () => {
    const component = shallow(<Cartridge nature="secondary" />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
