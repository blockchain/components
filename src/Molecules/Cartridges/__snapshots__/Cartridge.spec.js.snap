// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cartridge should render correctly 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFF;
  background-color: #004A7C;
  padding: 0.3rem;
  box-sizing: border-box;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Cartridge
    capitalize={false}
    italic={false}
    opacity={1}
    size="1rem"
    uppercase={false}
    weight={400}
  >
    <Styled(Text)
      capitalize={false}
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={400}
    >
      <Text
        capitalize={false}
        className="c0"
        italic={false}
        opacity={1}
        size="1rem"
        uppercase={false}
        weight={400}
      >
        <styled.span
          capitalize={false}
          className="c0"
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <span
            className="c0 c1"
            opacity={1}
            size="1rem"
          />
        </styled.span>
      </Text>
    </Styled(Text)>
  </Cartridge>
</ThemeProvider>
`;

exports[`Cartridge should render correctly with capitalize 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: capitalize;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFF;
  background-color: #004A7C;
  padding: 0.3rem;
  box-sizing: border-box;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Cartridge
    capitalize={true}
    italic={false}
    opacity={1}
    size="1rem"
    uppercase={false}
    weight={400}
  >
    <Styled(Text)
      capitalize={true}
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={400}
    >
      <Text
        capitalize={true}
        className="c0"
        italic={false}
        opacity={1}
        size="1rem"
        uppercase={false}
        weight={400}
      >
        <styled.span
          capitalize={true}
          className="c0"
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <span
            className="c0 c1"
            opacity={1}
            size="1rem"
          />
        </styled.span>
      </Text>
    </Styled(Text)>
  </Cartridge>
</ThemeProvider>
`;

exports[`Cartridge should render correctly with italic 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: italic;
  color: #545456;
  opacity: 1;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFF;
  background-color: #004A7C;
  padding: 0.3rem;
  box-sizing: border-box;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Cartridge
    capitalize={false}
    italic={true}
    opacity={1}
    size="1rem"
    uppercase={false}
    weight={400}
  >
    <Styled(Text)
      capitalize={false}
      italic={true}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={400}
    >
      <Text
        capitalize={false}
        className="c0"
        italic={true}
        opacity={1}
        size="1rem"
        uppercase={false}
        weight={400}
      >
        <styled.span
          capitalize={false}
          className="c0"
          italic={true}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <span
            className="c0 c1"
            opacity={1}
            size="1rem"
          />
        </styled.span>
      </Text>
    </Styled(Text)>
  </Cartridge>
</ThemeProvider>
`;

exports[`Cartridge should render correctly with opacity 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 0.5;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFF;
  background-color: #004A7C;
  padding: 0.3rem;
  box-sizing: border-box;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Cartridge
    capitalize={false}
    italic={false}
    opacity={0.5}
    size="1rem"
    uppercase={false}
    weight={400}
  >
    <Styled(Text)
      capitalize={false}
      italic={false}
      opacity={0.5}
      size="1rem"
      uppercase={false}
      weight={400}
    >
      <Text
        capitalize={false}
        className="c0"
        italic={false}
        opacity={0.5}
        size="1rem"
        uppercase={false}
        weight={400}
      >
        <styled.span
          capitalize={false}
          className="c0"
          italic={false}
          opacity={0.5}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <span
            className="c0 c1"
            opacity={0.5}
            size="1rem"
          />
        </styled.span>
      </Text>
    </Styled(Text)>
  </Cartridge>
</ThemeProvider>
`;

exports[`Cartridge should render correctly with size 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 18px;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFF;
  background-color: #004A7C;
  padding: 0.3rem;
  box-sizing: border-box;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Cartridge
    capitalize={false}
    italic={false}
    opacity={1}
    size="18px"
    uppercase={false}
    weight={400}
  >
    <Styled(Text)
      capitalize={false}
      italic={false}
      opacity={1}
      size="18px"
      uppercase={false}
      weight={400}
    >
      <Text
        capitalize={false}
        className="c0"
        italic={false}
        opacity={1}
        size="18px"
        uppercase={false}
        weight={400}
      >
        <styled.span
          capitalize={false}
          className="c0"
          italic={false}
          opacity={1}
          size="18px"
          uppercase={false}
          weight={400}
        >
          <span
            className="c0 c1"
            opacity={1}
            size="18px"
          />
        </styled.span>
      </Text>
    </Styled(Text)>
  </Cartridge>
</ThemeProvider>
`;

exports[`Cartridge should render correctly with uppercase 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: uppercase;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFF;
  background-color: #004A7C;
  padding: 0.3rem;
  box-sizing: border-box;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Cartridge
    capitalize={false}
    italic={false}
    opacity={1}
    size="1rem"
    uppercase={true}
    weight={400}
  >
    <Styled(Text)
      capitalize={false}
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={true}
      weight={400}
    >
      <Text
        capitalize={false}
        className="c0"
        italic={false}
        opacity={1}
        size="1rem"
        uppercase={true}
        weight={400}
      >
        <styled.span
          capitalize={false}
          className="c0"
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={true}
          weight={400}
        >
          <span
            className="c0 c1"
            opacity={1}
            size="1rem"
          />
        </styled.span>
      </Text>
    </Styled(Text)>
  </Cartridge>
</ThemeProvider>
`;
